lib/spack/spack/spec.py

```diff
@@ -3554,22 +3788,25 @@ def _dup(self, other, deps=True, cleardeps=True, caches=None):
         return changed
 
     def _dup_deps(self, other, deptypes, caches):
-        new_specs = {self.name: self}
-        for dspec in other.traverse_edges(cover='edges',
-                                          root=False):
-            if (dspec.deptypes and
-                not any(d in deptypes for d in dspec.deptypes)):
+        def spid(spec):
+            return id(spec)
```

Comment:
Why's this needed?  I think it's clearer with `id`

Reply:
It's a minor point, but what we need is something that is unique by spec during the "lifetime" of a copy. Using an indirect name means I can change it in a single line instead of having to go through multiple places where this is used.

Reply:
ok fair enough.  I think this will always be `id` though.